### PR TITLE
feat: Add "rebase all" functionality in master issue

### LINKED ISSUE #2765: Master Issue: Include "rebase all" option

### PR DESCRIPTION
**Pull request by hassan23**
An additional checkbox is added in the master issue under heading name **Rebase all**.
Checking the box will rebase all the **PRs** opened by **renovate Bot**.

After adding checkbox the master issue looks something like this:
Screen Shot 2019-04-25 at 7 03 39 PM

Closes #2765

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/process/index.js [hunk 1]
```diff
@@ -23,7 +23,22 @@ async function processRepo(config) {
           const [, type, branchName] = check.match(new RegExp(checkMatch));
           config.masterIssueChecks[branchName] = type;
         });
-        /* eslint-enable no-param-reassign */
+      }
+      const rebaseAllBranch = 'renovate/all';
+      if (rebaseAllBranch in config.masterIssueChecks) {
+        delete config.masterIssueChecks[rebaseAllBranch];
+        const unCheckMatch =
```
[the remark this answers is not in the document]
Comment (hassan23) [reply lib/workers/repository/process/index.js]: the logic is that if the user checked the `rebase all` then `renovate/all` branch gets included in the `config.masterIssueChecks` as a key. After this, I delete the `renovate/all` key and include every unchecked open PRs in `config.masterIssueChecks`.
But after your requested changes this logic will get changed a little bit.

File: lib/workers/repository/process/index.js [hunk 2]
```diff
@@ -23,6 +23,11 @@ async function processRepo(config) {
           const [, type, branchName] = check.match(new RegExp(checkMatch));
           config.masterIssueChecks[branchName] = type;
         });
+      }
+      const checkAllMatch = ' - \\[x\\] <!-- rebase-all-open-prs -->';
+      const checkedRebaseAll = issue.body.match(new RegExp(checkAllMatch, 'g'));
```
Comment (rarkins) [inline lib/workers/repository/process/index.js]: No need to match with 'g' here as there should only be one occurrence

File: lib/workers/repository/process/index.js [hunk 3]
```diff
@@ -23,6 +23,11 @@ async function processRepo(config) {
...
+      if (checkedRebaseAll && checkedRebaseAll.length) {
```
Comment (rarkins) [inline lib/workers/repository/process/index.js]: I think the length check is redundant - it should be `null` if there are no matches

File: lib/workers/repository/process/index.js [hunk 4]
```diff
@@ -23,6 +23,11 @@ async function processRepo(config) {
...
+        config.masterIssueRebaseAllOpen = 'rebase';
```
Comment (rarkins) [inline lib/workers/repository/process/index.js]: Set this to `true` instead of `"rebase"`

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/master-issue.js [hunk 5]
```diff
@@ -138,5 +138,14 @@ async function ensureMasterIssue(config, branches) {
     }
     issueBody += '\n';
   }
+  const alreadyOpened = branches.filter(branch => branch.res === 'done');
```
Comment (rarkins) [inline lib/workers/repository/master-issue.js]: Better to reuse this variable instead: https://github.com/renovatebot/renovate/blob/5e59756cf218d62fa7e0aa3a321dae92bf35d4e0/lib/workers/repository/master-issue.js#L113

Comment (rarkins) [reply lib/workers/repository/master-issue.js]: Test if inProgress.length > 1

Comment (hassan23) [reply lib/workers/repository/master-issue.js]: inProgress contain more than just the opened branches, it also contains the branches which is not created yet

Comment (rarkins) [reply lib/workers/repository/master-issue.js]: This is the array we use to populate the “Open” list right now so it seems like the best one to use here too

Comment (hassan23) [reply lib/workers/repository/master-issue.js]: ok, I will use this

**Comment by hassan23** [comment]
@rarkins Please review

**Comment by rarkins** [comment]
@hassan23 code looks good. I might experiment with the markdown/UI before merging though. Thanks!

**Pull request merged**